### ISSUE TITLE
addRoleForUser method not adding roles which were deleted before

### ISSUE THREAD

**Issue by apsnaidu** [casbin/node-casbin#193]
`addRoleForUser` method works fine when adding new roles/groups to policy database for the first time. However if we delete the roles/groups from the database manually or using `deleteRoleForUser` method and then try to add the roles again using the `addRoleForUser` method, it thinks that the roles already exist and doesn't add them again.

Below is the code snippet being run as part of our authentication flow:

```js
        await enforcer.addRoleForUser(sub, role))
          .then(async () => {
            await enforcer.enforce(sub, obj, act).then((pass) => {
              result = pass;
            });
          })
          .catch((err) => {
            console.log(err);
            throw err;
          });
        await enforcer.deleteRoleForUser(sub, role))
          .then(
            console.log(
              chalk.cyan.inverse(
                "Removed user from added AD groups in Casbin DB"
              )
            )
          )
          .catch((err) => {
            console.log(err);
            throw err;
          });
```
Basically we are adding the required AD groups for the user, authenticating him against the policy and then removing the roles from the user to make sure any roles removed in AD will also be removed from the policy database.
But with the current behaviour of `addRoleForUser` method we can't remove the roles and re-add them again 

Sequelize: 5.22.3
casbin: 5.1.3
casbin-sequelize-adapter: 2.1.0

**Comment by hsluoyz** [comment]
@zhmushan please take a look at this issue.

/cc @nodece

**Comment by ogabek96** [comment]
Works without problem when working with file. Will try to check with Sequelize

**Comment by ogabek96** [comment]
Hello @apsnaidu,
I wasn't able to repeat this error:
I tried to run the code like you described and it works fine as expected.
Here is my code:

`const casbin = require('casbin');
const { SequelizeAdapter } = require('casbin-sequelize-adapter');

async function myFunction() {

    const a = await SequelizeAdapter.newAdapter({
        username: 'casbin',
        password: '123456',
        database: 'casbin',
        dialect: 'postgres'
    });

    const enforcer = await casbin.newEnforcer(__dirname + '/casbin/rbac_model.conf', a);

    const sub = 'alice'; // the user that wants to access a resource.
    const obj = 'data1'; // the resource that is going to be accessed.
    const act = 'read'; // the operation that the user performs on the resource.
    const role = "role1"; // role

    await enforcer.getAllRoles()
        .then((res) => {
            console.log('roles before adding:', res);
        });
    // adding role
    await enforcer.addRoleForUser(sub, role)
        .then(async () => {
            await enforcer.getAllRoles()
                .then((res) => {
                    console.log('roles after adding', res);
                });
            await enforcer.enforce(sub, obj, act).then((pass) => {
                console.log('pass', pass)
            });
        })
        .catch((err) => {
            console.log(err);
            throw err;
        });
    //Deleting role
    await enforcer.deleteRoleForUser(sub, role)
        .then(
            console.log("Removed user from added AD groups in Casbin DB")
        )
        .catch((err) => {
            console.log(err);
            throw err;
        });

    //adding role after deleting
    await enforcer.addRoleForUser(sub, role)
        .then(async () => {
            await enforcer.getAllRoles()
                .then((res) => {
                    console.log('roles after deleting', res);
                });
        })
        .catch((err) => {
            console.log(err);
            throw err;
        });

}
myFunction();`

**Comment by hsluoyz** [comment]
@apsnaidu any update?

**Comment by hsluoyz** [comment]
Closed as stale.
